crates/ide/src/syntax_highlighting/highlight.rs

```diff
@@ -487,6 +489,8 @@ fn highlight_def(db: &RootDatabase, krate: Option<hir::Crate>, def: Definition)
             let ty = local.ty(db);
             if local.is_mut(db) || ty.is_mutable_reference() {
                 h |= HlMod::Mutable;
+            } else if local.is_ref(db) || ty.is_reference() {
+                h |= HlMod::Reference;
             }
```

Comment:
```suggestion
            } 
            if local.is_ref(db) || ty.is_reference() {
                h |= HlMod::Reference;
            }
```

let's do this here and elswhere, to keep mutability and reference-ness orthogonal then!